src/AzureClient/EntryPoint/EntryPoint.cs

```diff
@@ -34,18 +42,21 @@ internal class EntryPoint : IEntryPoint
         /// <param name="outputType">Specifies the output parameter type for the
         /// <see cref="EntryPointInfo{I,O}"/> object provided as the <c>entryPointInfo</c> argument.</param>
         /// <param name="operationInfo">Information about the Q# operation to be used as the entry point.</param>
+        /// <param name="qirStream">
+        ///     Stream from which QIR bitcode for the entry point can be read.
+        /// </param>
         /// <param name="logger">Logger used to report internal diagnostics.</param>
-        public EntryPoint(object entryPointInfo, Type inputType, Type outputType, OperationInfo operationInfo, ILogger? logger)
+        public EntryPoint(object entryPointInfo, Type inputType, Type outputType, OperationInfo operationInfo, Stream? qirStream, ILogger? logger)
         {
             EntryPointInfo = entryPointInfo;
             InputType = inputType;
             OutputType = outputType;
             OperationInfo = operationInfo;
             Logger = logger;
+            QirStream = qirStream;
         }
```

Comment:
(I'm fresh to this functionality and might not know something. I can be wrong)

I have an impression that this code is applicable to both the C# runtime and QIR runtime. 
For the C# runtime case the parameter `qirStream` is not applicable. If that's the case then it makes sense to make this param _last_ and to give this param a default value, such that in the C# case the caller can just skip passing the argument for this parameter:
>  public EntryPoint(object entryPointInfo, Type inputType, Type outputType, OperationInfo operationInfo, ILogger? logger **, Stream? qirStream = null**)

Reply:
This seems very reasonable. Done!